src/middleware/apiKeyChecker.ts

```diff
@@ -0,0 +1,23 @@
+import http from 'http';
+import { ApiKeyError } from '../errors';
+import type {
+  NextFunction,
+  Request,
+  Response,
+} from 'express';
+
+export const checkApiKey = (req: Request, res: Response, next: NextFunction): void => {
+  const apiKey = req.headers['x-api-key'] ?? '';
+  const authServer = process.env.AUTH_SERVER_URL ?? '';
+  const realmName = process.env.REALM_NAME ?? '';
+  http.get(`http://${authServer}/realms/${realmName}/check?apiKey=${apiKey.toString()}`, (authResponse: http.IncomingMessage) => {
```

Comment:
There is a lot going on in this line, and if the dev doesn't set those env vars it'll result in some confusing behavior -- I'd suggest:

1. Creating a utility function that returns the authServer base URL *AND* throws an error if either of the environment variables are set, (with a message explaining that a key environment variable isn't set).

2. Use [URLSearchParams](https://stackoverflow.com/a/49701878/159522) to define the querystring in that way rather than constructing the string manually (this becomes a bit more legible and also less error prone int he long run).